### DISCUSSION TITLE
custom authentication provider

### DISCUSSION THREAD

**Discussion by mai1015** [ory/kratos#3249] [Q&A]
Hello, im new to kratos. Is it possible to extend a new login flow like ethereum wallet signiture or other passwordless login in to the system? Thanks in advance.

**Comment by vinckr** [comment]
Hello @mai1015 
Sure! 
Ory Kratos already implements WebAuthn as passwordless option and Login via email/code is already possible with some workarounds and will be "properly" implemented soon as well.

> ethereum wallet signiture

I am not very familiar with the eth ecosystem but if it uses OIDC it should be easy to implement. 
Otherwise please add some more info around that or open a design document if you already have an idea how it could work.

**Reply by mai1015** [reply]
Thank you very much for your reply! So, the Ethereum wallet signature is basically an attempt to see if a user has access to a private key. The server requests an ECDSA signature on specific content from the user so that the server can validate the signature without the private key. Anyone who holds the private key will be able to do that.
